### PR TITLE
Bump Rust MSRV to 1.63.0

### PR DESCRIPTION
**Pull request by Nugine**
## Motivation and Context

I'm preparing a PR which upgrades `base64-simd` to v0.8.0. It needs Rust 1.63.0.

Previously: <https://github.com/awslabs/smithy-rs/pull/1938>

## Description

Bump Rust MSRV to 1.63.0

## Checklist

- [x] I have updated `CHANGELOG.next.toml` if I made changes to the smithy-rs codegen or runtime crates
- [x] I have updated `CHANGELOG.next.toml` if I made changes to the AWS SDK, generated SDK code, or SDK runtime crates

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: rust-runtime/aws-smithy-types/src/lib.rs [hunk 1]
```diff
@@ -12,6 +12,7 @@
     rust_2018_idioms,
     unreachable_pub
 )]
+#![allow(clippy::derive_partial_eq_without_eq)] // TODO: derive Eq for appropriate types
```
Comment (Velfi) [inline rust-runtime/aws-smithy-types/src/lib.rs:15]: Let's just add `Eq` to the derives for `Blob`

**Review by Velfi — CHANGES_REQUESTED** [review]

File: rust-runtime/aws-smithy-http-server-python/src/util.rs [hunk 2]
```diff
@@ -3,6 +3,8 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#![allow(clippy::derive_partial_eq_without_eq)] // TODO: derive Eq for appropriate types
```
[the remark this answers is not in the document]
Comment (Nugine) [reply rust-runtime/aws-smithy-http-server-python/src/util.rs]: Done.

File: rust-runtime/aws-smithy-xml/src/lib.rs [hunk 3]
```diff
@@ -5,6 +5,9 @@
 
 //! Abstractions for Smithy
 //! [XML Binding Traits](https://awslabs.github.io/smithy/1.0/spec/core/xml-traits.html)
+
+#![allow(clippy::derive_partial_eq_without_eq)] // TODO: derive Eq for appropriate types
```
[the remark this answers is not in the document]
Comment (Nugine) [reply rust-runtime/aws-smithy-xml/src/lib.rs]: Done

File: rust-runtime/aws-smithy-types/src/lib.rs (hunk 1 above)
Comment (Nugine) [reply rust-runtime/aws-smithy-types/src/lib.rs]: #2223 has added `Eq`. So I just allow the lint.

**Review by Velfi — APPROVED** [review]

**Comment by Velfi** [comment]
It looks like we're going to need to update codegen as well. I can take over from here and do that. Once that's done, it should be good to go.

**Comment by Velfi** [comment]
@Nugine I made the required updates on [this branch](https://github.com/awslabs/smithy-rs/tree/update/msrv-1.63-update-codegen-issues). Once you merge those changes into this branch, I can merge the PR.

**Comment by Nugine** [comment]
> @Nugine I made the required updates on [this branch](https://github.com/awslabs/smithy-rs/tree/update/msrv-1.63-update-codegen-issues). Once you merge those changes into this branch, I can merge the PR.

@Velfi Thanks. I cherry-picked your commit and rebased this PR on the main branch.

File: rust-runtime/aws-smithy-xml/src/decode.rs [hunk 4]
```diff
@@ -77,6 +77,7 @@ impl XmlDecodeError {
     }
 }
 
+#[allow(clippy::derive_partial_eq_without_eq)] // TODO: derive Eq?
```
Comment (Velfi) [inline rust-runtime/aws-smithy-xml/src/decode.rs:80]:
```suggestion
#[allow(clippy::derive_partial_eq_without_eq)]
```

File: rust-runtime/aws-smithy-types/src/timeout.rs [hunk 5]
```diff
@@ -276,6 +277,7 @@ impl TimeoutConfig {
 }
 
 /// Configuration subset of [`TimeoutConfig`] for operation timeouts
+#[allow(clippy::derive_partial_eq_without_eq)] // TODO: derive Eq?
```
Comment (Velfi) [inline rust-runtime/aws-smithy-types/src/timeout.rs:280]:
```suggestion
#[allow(clippy::derive_partial_eq_without_eq)]
```

File: rust-runtime/aws-smithy-types/src/timeout.rs [hunk 6]
```diff
@@ -199,6 +199,7 @@ impl From<TimeoutConfig> for TimeoutConfigBuilder {
 /// );
 /// # }
 /// ```
+#[allow(clippy::derive_partial_eq_without_eq)] // TODO: derive Eq?
```
Comment (Velfi) [inline rust-runtime/aws-smithy-types/src/timeout.rs:202]:
```suggestion
#[allow(clippy::derive_partial_eq_without_eq)]
```

File: rust-runtime/aws-smithy-types/src/retry.rs [hunk 7]
```diff
@@ -224,6 +225,7 @@ impl RetryConfigBuilder {
 }
 
 /// Retry configuration for requests.
+#[allow(clippy::derive_partial_eq_without_eq)] // TODO: derive Eq?
```
Comment (Velfi) [inline rust-runtime/aws-smithy-types/src/retry.rs:228]:
```suggestion
#[allow(clippy::derive_partial_eq_without_eq)]
```

File: rust-runtime/aws-smithy-types/src/retry.rs [hunk 8]
```diff
@@ -137,6 +137,7 @@ impl fmt::Display for RetryModeParseError {
 impl std::error::Error for RetryModeParseError {}
 
 /// Builder for [`RetryConfig`].
+#[allow(clippy::derive_partial_eq_without_eq)] // TODO: derive Eq?
```
Comment (Velfi) [inline rust-runtime/aws-smithy-types/src/retry.rs:140]:
```suggestion
#[allow(clippy::derive_partial_eq_without_eq)]
```

File: rust-runtime/aws-smithy-http-server-python/src/util.rs [hunk 9]
```diff
@@ -9,6 +9,7 @@ pub mod error;
 use pyo3::{PyAny, PyObject, PyResult, PyTypeInfo, Python};
 
 // Captures some information about a Python function.
+#[allow(clippy::derive_partial_eq_without_eq)] // TODO: derive Eq?
```
Comment (Velfi) [inline rust-runtime/aws-smithy-http-server-python/src/util.rs:12]:
```suggestion
#[allow(clippy::derive_partial_eq_without_eq)]
```

**Review by Velfi — CHANGES_REQUESTED** [review]
We have a lint for TODOs that don't reference an issue link. I don't see us adding `Eq` to these structs in the near future so just remove the TODOS. It should pass CI then, and I can merge it.

I decided to just update these myself, I hope you don't mind.

**Review by Velfi — DISMISSED** [review]

**Comment by Nugine** [comment]
It's still failing. Maybe this PR should wait for a separate fix about the Eq lint.

**Review by Velfi — APPROVED** [review]

**Review by jdisanti — APPROVED** [review]

**Pull request merged**